src/nbl/asset/utils/CSPIRVIntrospector.cpp

```diff
@@ -315,24 +315,25 @@ CSPIRVIntrospector::CStageIntrospectionData::SDescriptorVarInfo<true>* CSPIRVInt
 )
 {
     const uint32_t descSet = comp.get_decoration(r.id,spv::DecorationDescriptorSet);
-    if (descSet> DESCRIPTOR_SET_COUNT)
+    if (descSet > DESCRIPTOR_SET_COUNT)
         return nullptr; // TODO: log fail
 
     auto a = comp.get_name(r.id);
     const spirv_cross::SPIRType& type = comp.get_type(r.type_id);
-    const auto arrDim = type.array.size();
-    // assuming only 1D arrays because i don't know how desc set layout binding is constructed when it's let's say 2D array (e.g. uniform sampler2D smplr[4][5]; is it even legal?)
+    const size_t arrDim = type.array.size();
+    // only 1D arrays allowed
     if (arrDim>1)
         return nullptr; // TODO: log fail
 
-
+    const uint32_t descriptorArraySize = type.array.empty() ? 1u : *type.array.data();    
+    const bool isArrayTypeLiteral = descriptorArraySize <= 1u ? true : type.array_size_literal[0]; // can be set to true for runtime sized descriptors, doesn't make sense but also doesn't matter
```

Comment:
i think you have a bug here, a spec constant beign used would give you `descriptorArraySize = specConstantID`

then a spec constant Id of 0 or 1 would register as `isArrayTypeLiteral`

I think you should check `typ.array_size_literal[0]` first 

Or did you mean to check `arrDim!=0` here isntead?